### PR TITLE
Document the `X-Opaque-Id` header

### PR DESCRIPTION
**Pull request by andrross**
A [bug][1] was recently identified related to this header that was partly exascerbated by a misuse of the field. We have some documentation related to the [Tasks][2] API for this field, but no general guidance on how to use it.

[1]: https://github.com/opensearch-project/OpenSearch/issues/16702
[2]: https://opensearch.org/docs/latest/api-reference/tasks/#attaching-headers-to-tasks

### Version
All

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: _api-reference/common-parameters.md [hunk 1]
```diff
@@ -123,4 +123,17 @@ Kilometers | `km` or `kilometers`
...
+## `X-Opaque-Id` Header
+
+You can specify an opaque identifier on any request using the `X-Opaque-Id` header. This identifier is used to track tasks and deduplicate deprecation warnings in server side logs. This identifier is intended to differentiate callers into your OpenSearch cluster. Do not specify a unique value per request.
+
+#### Example request
+
+The following request adds an opaque ID to the request:
+
+```json
+
```
Comment (kolchfa-aws) [inline _api-reference/common-parameters.md:137]:
```suggestion
```

File: _api-reference/common-parameters.md [hunk 2]
```diff
@@ -123,4 +123,17 @@ Kilometers | `km` or `kilometers`
...
+curl -H "X-Opaque-Id: my-curl-client-1" -XGET localhost:9200/_tasks
+```
```
Comment (kolchfa-aws) [inline _api-reference/common-parameters.md:139]:
````suggestion
```
{% include copy.html %}
````

File: _api-reference/common-parameters.md [hunk 3]
```diff
@@ -123,4 +123,17 @@ Kilometers | `km` or `kilometers`
 Meters | `m` or `meters`
 Centimeters | `cm` or `centimeters`
 Millimeters | `mm` or `millimeters`
-Nautical miles | `NM`, `nmi`, or `nauticalmiles` 
\ No newline at end of file
+Nautical miles | `NM`, `nmi`, or `nauticalmiles` 
+
+## `X-Opaque-Id` Header
+
+You can specify an opaque identifier on any request using the `X-Opaque-Id` header. This identifier is used to track tasks and deduplicate deprecation warnings in server side logs. This identifier is intended to differentiate callers into your OpenSearch cluster. Do not specify a unique value per request.
```
Comment (kolchfa-aws) [inline _api-reference/common-parameters.md:130]:
```suggestion
You can specify an opaque identifier on any request using the `X-Opaque-Id` header. This identifier is used to track tasks and deduplicate deprecation warnings in server side logs. This identifier is intended to differentiate callers sending requests to your OpenSearch cluster. Do not specify a unique value per request.
```

**Review by kolchfa-aws — APPROVED** [review]
Thank you, @andrross!  A couple of suggestions.

File: _api-reference/common-parameters.md [hunk 4]
```diff
@@ -123,4 +123,17 @@ Kilometers | `km` or `kilometers`
 Meters | `m` or `meters`
 Centimeters | `cm` or `centimeters`
 Millimeters | `mm` or `millimeters`
-Nautical miles | `NM`, `nmi`, or `nauticalmiles` 
\ No newline at end of file
+Nautical miles | `NM`, `nmi`, or `nauticalmiles` 
+
+## `X-Opaque-Id` Header
```
Comment (natebower) [inline _api-reference/common-parameters.md:128]:
```suggestion
## `X-Opaque-Id` header
```

File: _api-reference/common-parameters.md [hunk 5]
```diff
@@ -123,4 +123,17 @@ Kilometers | `km` or `kilometers`
...
+
+You can specify an opaque identifier on any request using the `X-Opaque-Id` header. This identifier is used to track tasks and deduplicate deprecation warnings in server side logs. This identifier is intended to differentiate callers sending requests to your OpenSearch cluster. Do not specify a unique value per request.
```
Comment (natebower) [inline _api-reference/common-parameters.md:130]:
```suggestion
You can specify an opaque identifier for any request using the `X-Opaque-Id` header. This identifier is used to track tasks and deduplicate deprecation warnings in server-side logs. This identifier is used to differentiate between callers sending requests to your OpenSearch cluster. Do not specify a unique value per request.
```

**Review by natebower — COMMENTED** [review]
@kolchfa-aws @andrross Please see my changes and let me know if you have any questions. Thanks!

**Pull request merged**